src/main/java/bio/terra/workspace/service/workspace/flight/SyncSamGroupsStep.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.iam.model.IamRole;
+import bio.terra.workspace.service.job.JobMapKeys;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * A {@Code Step} which synchronizes Sam policies with google groups and stores the group names in
+ * the Stairway working map.
+ */
+public class SyncSamGroupsStep implements Step {
+
+  private final SamService samService;
+
+  @Autowired
```

Comment:
I don't think you can use @Autowired in steps. They are dynamically created by Stairway and won't get injection. I noticed it here, but you use it in other steps as well.

Reply:
Whoops, you're right you can't use `@Autowired` in steps. The step instances are really being created in the Flight, so those annotations aren't actually doing anything.